packages/eas-cli/src/update/configure.ts

```diff
@@ -252,6 +254,71 @@ async function ensureEASUpdatesIsConfiguredNativelyAsync(
   }
 }
 
+/**
+ * Make sure EAS Build profiles are configured to work with EAS Update by adding channels to build profiles.
+ */
+
+async function ensureEASUpdateIsConfiguredInEasConfigAsync({
+  projectDir,
+}: {
+  projectDir: string;
+}): Promise<void> {
+  const easConfigPath = EasJsonAccessor.formatEasJsonPath(projectDir);
+
+  if (await fs.pathExists(easConfigPath)) {
+    try {
+      const easConfigJson = await fs.readFile(easConfigPath, 'utf-8');
+      const easConfigData = JSON.parse(easConfigJson);
+      const easBuildProfilesWithChannels = Object.keys(easConfigData.build).reduce(
+        (acc, profileNameKey) => {
+          const buildProfile = easConfigData.build[profileNameKey];
+          const isNotAlreadyConfigured = !buildProfile.channel && !buildProfile.releaseChannel;
+
+          if (isNotAlreadyConfigured) {
+            return {
+              ...acc,
+              [profileNameKey]: {
+                ...buildProfile,
+                channel: profileNameKey,
+              },
+            };
+          }
+
+          return {
+            ...acc,
+            [profileNameKey]: {
+              ...easConfigData.build[profileNameKey],
+            },
+          };
+        },
+        {}
+      );
+
+      await fs.writeFile(
+        easConfigPath,
+        `${JSON.stringify(
+          {
+            ...easConfigData,
+            build: easBuildProfilesWithChannels,
+          },
+          null,
+          2
+        )}\n`
+      );
+
+      Log.withTick(`Configured ${chalk.bold('eas.json')}.`);
+    } catch (error) {
+      Log.error(`We were not able to configure ${chalk.bold('eas.json')}. Error: ${error}.`);
+    }
+  } else {
+    Log.warn(
+      `EAS Build is not configured. If you'd like to use EAS Build with EAS Update, run \`eas build:configure\`, then re-run \`eas update:configure\` to configure ${chalk.bold(
+        'eas.json'
+      )} with EAS Update.`
+    );
+  }
```

Comment:
eas.json supports JSON5. This doesn't work for JSON5 files. 
You should use https://github.com/expo/eas-cli/blob/main/packages/eas-json/src/accessor.ts#L91
Example of usage: https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/credentials/ios/IosCredentialsProvider.ts#L131